File: rules/csharp/tests/cs.weak-crypto/Crypto.cs
class CryptoTests {
  void Weak() {
    var a = MD5.Create();  // crit:expect cs.weak-crypto
    var b = SHA1.Create(); // crit:expect cs.weak-crypto
  }
  void Strong() {
    var c = SHA256.Create(); // crit:expect-not cs.weak-crypto
    var d = Aes.Create();    // crit:expect-not cs.weak-crypto
  }
}

File: rules/go/tests/go.weak-crypto/crypto.go
package main

import (
	"crypto/aes"
	"crypto/md5"
	"crypto/sha1"
	"crypto/sha256"
)

func weak() {
	_ = md5.New()  // crit:expect go.weak-crypto
	_ = sha1.New() // crit:expect go.weak-crypto
}

func strong(key []byte) {
	_ = sha256.New()       // crit:expect-not go.weak-crypto
	_, _ = aes.NewCipher(key) // crit:expect-not go.weak-crypto
}
